### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by ahegyes**
#### Changes proposed in this Pull Request

* Documentation updates based on open issues and PRs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,59 +1,120 @@
...
 ```
 
-## Scripts
+### Gutenberg blocks
+
+All custom-built blocks must be placed in a mu-plugin. The demo project contains a mu-plugin called `build-processes-demo-blocks` which exemplifies building two blocks called `build-processes-demo/foobar` and `build-processes-demo/spamham`, respectively.
+
+By separating the blocks from the theme into a mu-plugin, we can ensure that the blocks are always available on the site, even if the theme is changed (e.g., because of a future redesign or as part of the debugging process). Moreover, it forces us, as developers, to think about the blocks as a separate entity from the theme and to design the code as such thus making it easier to reuse them in other projects.
+
+The mu-plugin must contain enough CSS for a basic styling of the blocks, but it's fine if the theme provides its own CSS for maintaining the site's look-and-feel of the site. Take [Seedlet as an example](https://github.com/Automattic/themes/tree/trunk/seedlet/assets/sass/blocks) of the theme providing its own styling for the blocks.
```
Comment (tommusrhodus) [inline README.md]: > The mu-plugin must contain enough CSS for a basic styling of the blocks,

Perhaps this is better worded as something like:

> The mu-plugin must contain enough CSS for the block to be functional and not appear broken. All other styling can be held in the mu-plugin or the theme.

That way we make it clear that we want the block to be working accross projects, but it doesn't have to look "good" out of the box on another project.

File: README.md [hunk 2]
```diff
@@ -1,59 +1,120 @@
...
-This can be used to install a parent theme or a plugin that is required for the theme to work.
+It's possible to install themes and plugins which are available on the WordPress.org repository through the `composer.json` file. This can help with development and debugging as it allows us to install a parent theme or a plugin that is required for the theme to work without tracking it through `git`.
+
+Installing non-WP.org themes and plugins via Composer is also possible if the plugin offers support for it. For example, all plugins sold through Freemius do support Composer, as well as some popular ones such as Advanced Custom Fields Pro.
 
-### Tips & Tricks (not required)
+### Installing WordPress itself as a Composer dependency
 
-You can include WordPress as a Composer dev dependency by including the package `johnpbloch/wordpress-core` as a Composer dev dependency.
+You can install WordPress itself as a Composer dependency by including the package `johnpbloch/wordpress-core`. This can be useful for PHPStorm's code completion or simply for searching through the codebase locally.
```
Comment (tommusrhodus) [inline README.md:80]: Could we highlight that this is optional as is offered as a "nice to have" only.

**Comment by tommusrhodus** [comment]
@ahegyes This is amazing! 🤩

**Comment by ahegyes** [comment]
Approved in meeting from Feb 22nd

**Pull request merged**